llvm/lib/Target/NVPTX/NVPTXLowerArgs.cpp

```diff
@@ -641,7 +629,11 @@ void NVPTXLowerArgs::handleByValParam(const NVPTXTargetMachine &TM,
     copyByValParam(*Func, *Arg);
 }
 
-void NVPTXLowerArgs::markPointerAsGlobal(Value *Ptr) {
+// Knowing Ptr must point to the global address space, this function
+// addrspacecasts Ptr to global and then back to generic. This allows
+// NVPTXInferAddressSpaces to fold the global-to-generic cast into
+// loads/stores that appear later.
```

Comment:
This could use some editing. E.g. 
```
If generic `Ptr` is known to point to global memory, add ASC to global AS and back to generic. 
It will allow us to lower load/store operations on it via `{ld,st}.global`
```